File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    var answer = [0,0];
    let result = 0;
    let str = s;
    
    while(result != 1) {
        let temp = str;
        str = str.split("").filter((num) => num != 0).join("");
        result = Number(str.length).toString(2);
        answer[0]++;
        answer[1] += temp.length - str.length;
        str = result;
    }
    return answer;
}

File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
import java.util.*;
class Solution {
    public int[] solution(String s) {
    int[] answer = {0,0};
    String result = "0";
    String str = s;

    while(!result.equals("1")) {
      String temp = str;
      String[] strArr= Arrays.stream(str.split("")).filter(i -> i.equals("1")).toArray(String[]::new);
      str = String.join("",strArr);
      result = Integer.toBinaryString(str.length());
      answer[0]++;
      answer[1] += temp.length() - str.length();
      str = result;

    }
    return answer;
    }
}